helm/resource_release.go

```diff
@@ -1508,3 +1513,16 @@ func valuesKnown(d *schema.ResourceDiff) bool {
 	}
 	return true
 }
+
+func useChartVersion(chart string, repo string) bool {
+	_, urlerr := url.ParseRequestURI(chart)
+
+	if _, err := os.Stat(chart); err == nil || urlerr == nil {
```

Comment:
Because this conditional is an or, you can simply have a separate if statement above that returns true for the `urlerr` condition. 

It's easier to understand if you formulate the logic as separate conditionals that return early.